packages/workflow/src/utils.ts

```diff
@@ -71,14 +71,28 @@ export const deepCopy = <T extends ((object | Date) & { toJSON?: () => string })
 type MutuallyExclusive<T, U> =
 	| (T & { [k in Exclude<keyof U, keyof T>]?: never })
 	| (U & { [k in Exclude<keyof T, keyof U>]?: never });
+type JSONParseOptions<T> = MutuallyExclusive<
+	{ errorMessage?: string },
+	{ fallbackValue?: T | (() => T) }
+>;
 
-type JSONParseOptions<T> = MutuallyExclusive<{ errorMessage: string }, { fallbackValue: T }>;
-
+/**
+ * Parses a JSON string into an object with optional error handling and recovery mechanisms.
+ *
+ * @param {string} jsonString - The JSON string to parse.
+ * @param {Object} [options] - Optional settings for parsing the JSON string. Either `fallbackValue` or `errorMessage` can be set, but not both.
+ * @param {string} [options.errorMessage] - A custom error message to throw if the JSON string cannot be parsed.
+ * @param {*} [options.fallbackValue] - A fallback value to return if the JSON string cannot be parsed.
+ * @returns {Object} - The parsed object, or the fallback value if parsing fails and `fallbackValue` is set.
+ */
 export const jsonParse = <T>(jsonString: string, options?: JSONParseOptions<T>): T => {
 	try {
 		return JSON.parse(jsonString) as T;
 	} catch (error) {
 		if (options?.fallbackValue !== undefined) {
+			if (options.fallbackValue instanceof Function) {
```

Comment:
I do not like this approach, now instead of passing simple boolean option its expected  that user of this helper had to setup their own fallback function, I would prefer that we return back to solution with option